hips/hip-9999.md

```diff
@@ -0,0 +1,240 @@
+---
+hip: 9999
+title: "New annotations for displaying job output"
+authors: [ "Ian Zink <ian@replicated.com>" ]
+created: "2023-01-26"
+type: "feature"
+status: "draft"
+---
+
+## Abstract
+
+This proposes a new annotation to indicate that the output from the job should be displayed to the user.
+
+## Motivation
+
+The primary motivation for this HIP is the ability to display Preflight checks before the main chart installs. This provides vital feedback to the user as to why the chart install is not proceeding.
+
+Often it is important to verify that the kubernetes cluster you are deploying a helm chart into has certain properties. You might need to know that the cluster is of a certain version to use various APIs. You might need to know that it has ingress available, a certain amount of ephemeral storage, memory, or CPUs available. You might want to validate the the service key they provided was correct or that that database they entered is reachable. Letting a chart deploy and then finding debugging to see why it failed is a poor user experience. These things can all be done with preflight checks enabled by the hook proposed in this HIP.
```

Comment:
For the first, I know that Helm does provide the ability to check for api versions, but there might be good reasons you want to declare it in a tool specifically designed to declare and report on the readiness of a cluster for your app. But for the purposes of this HIP I do see where it could be confusing so I'll just take it out as it's not really crucial here anyway.

For the second, while you can check if an ingress api is available, as far as I'm aware there is not a way to tell if there is a controller that will do anything with it -- which is what this is referring to so I'll be more specific.

Reply:
I'm going to resolve this let me know if you think it still needs clarification.